src/main/java/uk/gov/hmcts/reform/prl/controllers/ManageOrdersController.java

```diff
@@ -0,0 +1,153 @@
+package uk.gov.hmcts.reform.prl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CallbackResponse;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+import uk.gov.hmcts.reform.prl.models.language.DocumentLanguage;
+import uk.gov.hmcts.reform.prl.services.DocumentLanguageService;
+import uk.gov.hmcts.reform.prl.services.ManageOrderEmailService;
+import uk.gov.hmcts.reform.prl.services.ManageOrderService;
+import uk.gov.hmcts.reform.prl.services.UserService;
+import uk.gov.hmcts.reform.prl.utils.CaseUtils;
+
+import java.util.Map;
+import javax.ws.rs.core.HttpHeaders;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+@RestController
+@RequiredArgsConstructor
+public class ManageOrdersController {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private final UserService userService;
+
+    @Autowired
+    private  ManageOrderService manageOrderService;
+
+    @Autowired
+    private final DocumentLanguageService documentLanguageService;
+
+    @Autowired
+    private ManageOrderEmailService manageOrderEmailService;
+
+    @Value("${document.templates.common.prl_c21_template}")
+    protected String c21Template;
+
+    @Value("${document.templates.common.prl_c21_filename}")
+    protected String c21File;
+
+    @Value("${document.templates.common.prl_c21_welsh_template}")
+    protected String c21WelshTemplate;
+
+    @Value("${document.templates.common.prl_c21_welsh_filename}")
+    protected String c21WelshFile;
+
+    @PostMapping(path = "/populate-preview-order", consumes = APPLICATION_JSON, produces = APPLICATION_JSON)
+    @ApiOperation(value = "Callback to show preview order in next screen for upload order")
+    public AboutToStartOrSubmitCallbackResponse populatePreviewOrderWhenOrderUploaded(
+        @RequestHeader(org.springframework.http.HttpHeaders.AUTHORIZATION) String authorisation,
+        @RequestBody CallbackRequest callbackRequest) throws Exception {
+        CaseData caseData1 = CaseUtils.getCaseData(callbackRequest.getCaseDetails(), objectMapper);
```

Comment:
Need a proper variable name

Reply:
changing this will cause change in ccd code too